src/RateLimiter/Counter.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Web\RateLimiter;
+
+use Psr\SimpleCache\CacheInterface;
+
+/**
+ * Counter implements generic сell rate limit algorithm (GCRA) that ensures that after reaching the limit futher
+ * increments are distributed equally.
+ *
+ * @link https://en.wikipedia.org/wiki/Generic_cell_rate_algorithm
+ */
+final class Counter implements CounterInterface
+{
+    public const ID_PREFIX = 'rate-limiter-';
```

Comment:
Maybe for debuggin and tests add the method `getCacheKey()`  and set this constant `private`?
Later it may used in any services as `$key = $this->counter->getCacheKey()` instead of `$key = Counter::ID_PREFIX . ?`

Reply:
Done.